packages/compass-components/src/components/document-list/element.tsx

```diff
@@ -567,6 +572,9 @@ export const HadronElement: React.FunctionComponent<{
             </div>
           </div>
         )}
+        {typeof extraGutterWidth === 'number' && (
+          <div style={{ width: extraGutterWidth }} />
+        )}
         <div className={elementSpacer} style={{ width: elementSpacerWidth }}>
```

Comment:
Nit: this element is known to be a performance bottleneck due to the sheer amount of dom elements it might end up rendering on the page (yeah, even if they are just empty divs), so one thing I'd suggest here is to combine your extra width into the existing element:

```suggestion
        <div className={elementSpacer} style={{ width: elementSpacerWidth + (extraGutterWidth ?? 0) }}>
```

(or you can include it in the existing calculate extra width function above)

It's less of an issue now that we're virtualizing the list of these document cards in most cases, but doesn't hurt to be a bit more vigilant about that in this particular element.

Also if you want to keep the extra element, you might want to add the same elementSpacer class to it to make sure that it won't get the default flex shrink value that would allow it to be resized inside of this flex parent

Reply:
I've updated the code to use the existing spacer and fixed what I believe was a bug in the calculation of the spacer width 🤞